pilot/pkg/xds/filters/filters.go

```diff
@@ -300,6 +278,58 @@ var (
 	}
 )
 
+func BuildSidecarOutboundMetadataFilter(skipHeaders bool) *hcm.HttpFilter {
+	// TODO GregHanson
+	// false values can be omitted in protobuf, results in diff JSON values between controlplane and envoy config dumps
+	// long term fix will be to add the metadata config to istio/api and use that over TypedStruct
+	if skipHeaders {
+		return &hcm.HttpFilter{
+			Name: MxFilterName,
+			ConfigType: &hcm.HttpFilter_TypedConfig{
+				TypedConfig: protoconv.TypedStructWithFields("type.googleapis.com/io.istio.http.peer_metadata.Config",
+					map[string]any{
+						"upstream_discovery": []any{
+							map[string]any{
+								"istio_headers": map[string]any{},
+							},
+							map[string]any{
+								"workload_discovery": map[string]any{},
+							},
+						},
+						"upstream_propagation": []any{
+							map[string]any{
+								"istio_headers": map[string]any{
+									"skip_external_clusters": true,
```

Comment:
there is a difference in how `:15014/debug/config_dump` on istiod displays this TypedStruct and how envoy config_dump displays it.  when false, it's omitted in one.  when skip_external_clusters is set to false, istiod returns the expected json object:
```
                        "httpFilters": [
                          {
                            "name": "istio.metadata_exchange",
                            "typedConfig": {
                              "@type": "type.googleapis.com/udpa.type.v1.TypedStruct",
                              "typeUrl": "type.googleapis.com/io.istio.http.peer_metadata.Config",
                              "value": {
                                "upstream_discovery": [
                                  {
                                    "istio_headers": {}
                                  },
                                  {
                                    "workload_discovery": {}
                                  }
                                ],
                                "upstream_propagation": [
                                  {
                                    "istio_headers": {
                                      "skip_external_clusters": false
                                    }
                                  }
                                ]
                              }
                            }
                          },
```
but envoy's config dump is omitting the false value in it's config_dump:
```
                            "httpFilters": [
                                {
                                    "name": "istio.metadata_exchange",
                                    "typedConfig": {
                                        "@type": "type.googleapis.com/udpa.type.v1.TypedStruct",
                                        "typeUrl": "type.googleapis.com/io.istio.http.peer_metadata.Config",
                                        "value": {
                                            "upstream_discovery": [
                                                {
                                                    "istio_headers": {}
                                                },
                                                {
                                                    "workload_discovery": {}
                                                }
                                            ],
                                            "upstream_propagation": [
                                                {
                                                    "istio_headers": {}
                                                }
                                            ]
                                        }
                                    }
                                },
```
leading to failures in proxy-status integ tests since there is a diff in the json:
https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/46579/integ-pilot_istio/1696301170378674176

when `true` things behave as expected

Reply:
for now, getting around it by using different pre-built filter objects, which was something suggested earlier to optimize performance